pytorch_lightning/trainer/trainer.py

```diff
@@ -1158,10 +1163,8 @@ def _run(
 
         self.checkpoint_connector.resume_end()
 
-        # dispatch `start_training` or `start_evaluating` or `start_predicting`
-        results = self._dispatch()
-
-        self._post_dispatch()
+        results = self.run_stage()
```

Comment:
nit: run_stage could be marked private as it's not used outside the class. For another PR 😀

Reply:
Good that you mention this. I looked at usages and found HorovodPlugin is the only one left that calls `trainer.run_stage()`. I completely missed this until now. Hashing it out in #10989 as a prerequisite for the PR here.